sdk/src/main/java/com/karhoo/sdk/analytics/Payloader.kt

```diff
@@ -157,6 +238,101 @@ class Payloader internal constructor(builder: Builder) {
             return this
         }
 
+        fun checkoutOpened(quoteId: String?): Builder {
+            quoteId?.let {
+                payload[QUOTE_ID] = it
+            }
+
+            return this
+        }
+
+        fun cardAuthorisationSuccess(quoteId: String?): Builder {
+            quoteId?.let {
+                payload[QUOTE_ID] = it
+            }
+
+            return this
+        }
+
+        fun loyaltyStatusRequested(
+            slug: String?,
+            quoteId: String?,
+            correlationId: String?,
+            loyaltyStatus: LoyaltyStatus?,
+            balance: Int?,
+            loyaltyProgramme: LoyaltyProgramme?
+        ): Builder {
+            payload[LOYALTY_ENABLED] =
+                loyaltyStatus?.canBurn == false && loyaltyStatus.canEarn == false
+            payload[LOYALTY_STATUS_SUCCESS] = loyaltyStatus != null
+            payload[LOYALTY_STATUS_CAN_BURN] = loyaltyStatus?.canBurn ?: false
+            payload[LOYALTY_STATUS_CAN_EARN] = loyaltyStatus?.canEarn ?: false
+            payload[LOYALTY_STATUS_BALANCE] = balance ?: 0
+
+            quoteId?.let {
+                payload[QUOTE_ID] = it
+            }
+            loyaltyProgramme?.name?.let {
+                payload[LOYALTY_NAME] = it
+            }
+            correlationId?.let {
+                payload[CORRELATION_ID] = it
+            }
+            slug?.let {
+                payload[LOYALTY_STATUS_ERROR_SLUG] = it
+            }
+
+            return this
+        }
+
+        fun loyaltyPreAuthFailure(
+            slug: String?,
+            quoteId: String?,
+            correlationId: String?,
+            loyaltyMode: String,
+            balance: Int?
+        ): Builder {
+            quoteId?.let {
+                payload[QUOTE_ID] = it
+            }
+            correlationId?.let {
+                payload[CORRELATION_ID] = it
+            }
+            slug?.let {
+                payload[LOYALTY_STATUS_ERROR_SLUG] = it
+            }
+            balance?.let {
+                payload[LOYALTY_STATUS_BALANCE] = it
+            }
+
+            payload[LOYALTY_PREAUTH_TYPE] = loyaltyMode
+
+            return this
+        }
+
+        fun loyaltyPreAuthSuccess(
+            quoteId: String?,
+            correlationId: String?,
+            loyaltyMode: String?,
+            balance: Int?
+        ): Builder {
+            quoteId?.let {
+                payload[QUOTE_ID] = it
+            }
+            correlationId?.let {
+                payload[CORRELATION_ID] = it
+            }
+            loyaltyMode?.let {
+                payload[LOYALTY_PREAUTH_TYPE] = loyaltyMode
+            }
+
+            balance?.let {
```

Comment:
extra info on balance compared to the spreadsheet requirements. @BartlomiejSopalaKarhoo do we also send this on iOS?